### DISCUSSION TITLE
Error when running ww3_shel

### DISCUSSION THREAD

**Discussion by abayhaqi** [NOAA-EMC/WW3#827] [Q&A] [answered]
Hi , 
Please help me, I have a following message when I run ww3_shel. I would really appreciate if anyone helps me to solve this problem. Thank you. 

                     *** WAVEWATCH III Program shell ***
               ===============================================

  Comment character is '$'

At line 1219 of file /home/bayhaqi/WW3/model/src/ww3_shel.F90 (unit = 10, file = './ww3_shel.inp')
Fortran runtime error: Bad logical value while reading item 2

Error termination. Backtrace:
#0  0x7f56b3c50d21 in ???
#1  0x7f56b3c51869 in ???
#2  0x7f56b3c5254f in ???
#3  0x7f56b3e8d74e in ???
#4  0x7f56b3e8f179 in ???
#5  0x7f56b3e8fdc9 in ???
#6  0x7f56b3f52e94 in w3shel
        at /home/bayhaqi/WW3/model/src/ww3_shel.F90:1219
#7  0x7f56b3f526ae in main
        at /home/bayhaqi/WW3/model/src/ww3_shel.F90:255

**Comment by JessicaMeixner-NOAA** [comment, marked as answer]
@abayhaqi looks like there's an issue with our ww3_shel.inp file.  Can you share that?

**Reply by abayhaqi** [reply]
Hi @JessicaMeixner-NOAA , Thank you for your response. 
Herewith I attach my ww3_shep.inp file.
Looking forward to hearing the feedback soon.

Thanks

[ww3_shel.inp.txt](https://github.com/NOAA-EMC/WW3/files/9820499/ww3_shel.inp.txt)

**Reply by JessicaMeixner-NOAA** [reply]
What version of the code are you using? 

Two thinigs that might help would be to use namelist variables for your output selection, see: 
https://github.com/NOAA-EMC/WW3/blob/develop/regtests/ww3_tp1.10/input/ww3_shel.inp#L20-L21 as an example of this. 

Additionally please see: https://github.com/NOAA-EMC/WW3/wiki/FAQs-page#how-to-debug-error-messages

**Reply by abayhaqi** [reply]
@JessicaMeixner-NOAA I use v.6.07, but the same error massage still appears

**Reply by JessicaMeixner-NOAA** [reply]
Try updating your ww3_shel from: 

```
$ (1) Forcing Fields
  T
$ DPT CUR WND AST WLV ICE IBG D50
$ T   T   T   T   T   F   F   F
```

to: 
```
$ (1) Forcing Fields
  T
$ DPT CUR WND AST WLV ICE IBG D50
 T   T   T   T   T   F   F   F
```

**Reply by abayhaqi** [reply]
@JessicaMeixner-NOAA , Yapp, I have tried that. I also tried with only set the standard mean mean parameters as the output (like the attachment file). However, still same error massage appears. 

[ww3_shel.inp.txt](https://github.com/NOAA-EMC/WW3/files/9823317/ww3_shel.inp.txt)

**Reply by JessicaMeixner-NOAA** [reply]
For version 6.07 you should just need: 
```
   F F     Water levels
   F F     Currents
   T F     Winds
   F       Ice concentrations
   F       Assimilation data : Mean parameters
   F       Assimilation data : 1-D spectra
   F       Assimilation data : 2-D spectra.
```

**Reply by abayhaqi** [reply]
@JessicaMeixner-NOAA Thank you for mentioning that. I am sorry if I ask too much, but I get another error message. I already put T for the output group 2 (standard mean wave parameters). Is there another thing that I should revise? Thank you


                  *** WAVEWATCH III Program shell ***
               ===============================================

  Comment character is '$'


  Input fields :
 --------------------------------------------------
       water levels   ---/NO
       currents       ---/NO
       winds          YES/--
       ice fields     ---/NO
       momentum       ---/NO
       air density    ---/NO
       mean param.    ---/NO
       1D spectra     ---/NO
       2D spectra     ---/NO


 *** WAVEWATCH III WARNING  :
     NUMBER OF REQUESTED OUTPUT FIELD FLAGS IN GROUP  2
 LESS THAN AVAILABLE, CHECK DOCS FOR MORE OPTIONS
            Fields   : Wave height
                       Mean wave length
                       Mean wave period(+2)
                       Mean wave period(-1)
                       Mean wave period(+1)
                       Peak frequency
                       Mean wave dir. a1b1
                       Mean dir. spr. a1b1
                       Peak direction

 *** WAVEWATCH III ERROR IN NEXTLN :
     PREMATURE END OF INPUT FILE


EXTCDE MPI_ABORT, IEXIT=     1

--------------------------------------------------------------------------
MPI_ABORT was invoked on rank 0 in communicator MPI_COMM_WORLD
with errorcode 1.

NOTE: invoking MPI_ABORT causes Open MPI to kill all MPI processes.
You may or may not see output from other processes, depending on
exactly when Open MPI kills them.

**Reply by JessicaMeixner-NOAA** [reply]
@abayhaqi your output requests seem to not be matching.  The easiest fix for this namelist instead of the T/F lists as you are doing now.  Please see, for example: https://github.com/NOAA-EMC/WW3/blob/6.07/regtests/mww3_test_01/input/ww3_shel.inp#L17-L20

**Reply by abayhaqi** [reply]
@JessicaMeixner-NOAA Hi, It works well now. Thank you so much for helping me.

**Reply by abayhaqi** [reply]
@JessicaMeixner-NOAA ,
Hi Jessica, there is another problem. I successfully ran the simulation. I ran in a month with the setting of hourly output. So I expect that the output will be 745 files. In fact, when I converted the ww3.ctl to the nc file, the results were only 721 files. 
There is a missing time step/file. 
Do you have any ideas to solve this matter?

thank you
